### DISCUSSION TITLE
Pass empty parameter to LearningRuleType

### DISCUSSION THREAD

**Discussion by cnrgbot** [nengo/nengo-zoo#766] [Backends]
> _Archived from the Nengo Discourse forum._ Originally posted by **Tioz** on 2020-08-19. (Forum topic #1346, post #1 / id 3654)

Is there any way of passing an empty parameter to a LearningRuleType?  This would be useful, for example, if trying to run a model in a script taking command-line parameters.  If no command-line parameter were specified (for ex. the PES learning rate) I would like the learning rule to use the default for its parameter:
> NumberParam("learning_rate", low=0, readonly=True, default=1e-4)

I would then be able to instantiate the object in my script in a way similar to this:
> conn.learning_rule_type = PES(learning_rate=empty())

```None``` does not work with non-optional parameters and ```DefaultParameterType``` also is not accepted.

**Comment by cnrgbot** [comment]
> _Archived from the Nengo Discourse forum._ Originally posted by **xchoo** on 2020-08-19. (Forum topic #1346, post #2 / id 3656)

Most (if not all) Nengo objects can be instantiated with default values using the `Default` class from `nengo.params`. Here's some example code specific to your question:
```
from nengo.params import Default

conn.learning_rule_type = PES(learning_rate=Default)
```
